rs/nervous_system/root/src/change_canister.rs

```diff
@@ -256,24 +304,51 @@ async fn install_code(request: ChangeCanisterRequest) -> ic_cdk::api::call::Call
     let canister_id = canister_id.get();
     let sender_canister_version = Some(ic_cdk::api::canister_version());
 
-    let install_code_args = InstallCodeArgs {
-        mode,
-        canister_id,
-        wasm_module,
-        arg,
-        compute_allocation,
-        memory_allocation,
-        sender_canister_version,
-    };
-    // Warning: despite dfn_core::call returning a Result, it actually traps when
-    // the callee traps! Use the public cdk instead, which does not have this
-    // issue.
-    ic_cdk::api::call::call(
-        Principal::try_from(IC_00.get().as_slice()).unwrap(),
-        "install_code",
-        (&install_code_args,),
-    )
-    .await
+    if let Some(ChunkedCanisterWasm {
```

Comment:
There will need to be validation somewhere that chunked_canister_wasm and wasm_module are mutually exclusive, I think.

Reply:
I was thinking about it, too, but couldn't convince myself that this was necessary. 

Currently, if there are some bytes in `wasm_module` and `chunked_canister_wasm`, then the former would be used and the latter ignored. 

Are you sure it's better to reject requests that specify more data than we need? 

Either way, I'm happy to add a unit test to make sure this behavior is well specified.

Reply:
I think the principle of least surprise should apply.  One would not expect to be able to submit two mutually exclusive options into an 
API and not get an error.  If the API silently ignores one, this could be confusing.  

It might be a good change, for future forward compatibility, to make the new field actually support both as an enum, and then translate the old field into the enum as part of the validation, so that we can eventually retire the old field and have the type structure enforce the either-or nature of the arguments.